example-utils/integration-test-utils/src/integration_test_utils.rs

```diff
@@ -0,0 +1,344 @@
+                            "{} set connected_to_streamer to: {:?}",
+                            &name, active_connection_listing
+                        );
+                    }
+                    ClientCommand::DisconnectedFromStreamer(active_connections) => {
+                        debug!("{} commmand: DisconnectedFromStreamer", &name);
+                        *active_connection_listing = active_connections.clone();
+                        debug!(
+                            "{} set connected_to_streamer to: {:?}",
+                            &name, active_connection_listing
+                        );
+                    }
+                    _ => {
+                        error!(
+                            "{} ClientCommand::Stop should have been handled earlier",
+                            &name
+                        )
+                    }
+                }
+                {
+                    let (lock_exec_pause, cvar_exec_pause) = &*execution_paused;
+                    let mut has_paused = lock_exec_pause.lock().await;
+                    *has_paused = true;
+                    debug!("{} has_paused set to true", &name);
+                    cvar_exec_pause.notify_one();
+                    debug!("{} cvar_exec_pause.notify_one()", &name);
+                }
+                debug!("{} Loop paused. Waiting...", &name);
+                should_pause = cvar.wait(should_pause).await;
+                debug!("{} Got signal to pause", &name);
+            }
+        }
+
+        false
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn send_message_set(
+    client: &UPClientFoo,
+    name: &str,
+    client_rand_b: u64,
+    msg_type: &str,
+    msg_set: &mut [UMessage],
+    active_connection_listing: &ActiveConnections,
+    sent_messages: &mut Vec<UMessage>,
+    number_of_sends: Arc<AtomicU64>,
+) {
+    for (index, msg) in &mut msg_set.iter_mut().enumerate() {
+        if let Some(attributes) = msg.attributes.as_mut() {
+            let new_id = UUIDBuilder::build();
+            attributes.id.0 = Some(Box::new(new_id));
+            let uuid = attributes.id.as_mut().unwrap();
+            let lsb = &mut uuid.lsb;
+            *lsb = override_lsb_rand_b(*lsb, client_rand_b);
+        }
+
+        debug!(
+            "prior to sending from client {}, the {msg_type} message: {:?}",
+            &name, &msg
+        );
+
+        let send_res = client.send(msg.clone()).await;
+        if send_res.is_err() {
+            error!("Unable to send from client: {}", &name);
+        } else if !active_connection_listing.is_empty() && active_connection_listing[index] {
+            sent_messages.push(msg.clone());
+            number_of_sends.fetch_add(1, Ordering::SeqCst);
+            debug!(
+                "{} after {msg_type} send, we have sent: {}",
+                &name,
+                number_of_sends.load(Ordering::SeqCst)
+            );
+        }
+    }
+}
+
+pub async fn run_client(
```

Comment:
Much cleaner now. :+1: 
One question here: Why do we need ClientControl to control streamer in the tests? I haven't taken a deeper look, but will it cause any issues if we don't use the control?

Reply:
Normally in a production use case we wouldn't need to do what I'm doing here.

I'm doing some careful orchestration here so that we can get exactly matching counts of sends vs receives to make the testing easier.